### ISSUE TITLE
training problem

### ISSUE THREAD

**Issue by fullstack-dev0727** [lifeiteng/vall-e#86]
Hi, @lifeiteng 
I have tried to train with libritts dataset on 8 x GPUs. (main branch)
I changed world-size into 8 as default.

Command: 
python3 bin/trainer.py --max-duration 40 --filter-min-duration 0.5 --filter-max-duration 14     --model-name "VALL-E" --norm-first true --add-prenet false --dtype "float32"     --decoder-dim 1024 --nhead 16 --num-decoder-layers 12 --prefix-mode 1     --base-lr 0.05 --warmup-steps 200 --average-period 0 --accumulate-grad-steps 1     --num-epochs 201 --start-epoch 1 --start-batch 0     --exp-dir exp/valle 

But I am getting following logs for a few days.
2023-04-13 16:00:05,621 INFO [trainer.py:650] (5/8) Reaches end of dataloader.
2023-04-13 16:00:05,622 INFO [trainer.py:626] (5/8) Computing validation loss
2023-04-13 16:00:05,629 INFO [trainer.py:650] (7/8) Reaches end of dataloader.
2023-04-13 16:00:05,630 INFO [trainer.py:626] (7/8) Computing validation loss
2023-04-13 16:00:05,636 INFO [trainer.py:650] (4/8) Reaches end of dataloader.
2023-04-13 16:00:05,637 INFO [trainer.py:626] (4/8) Computing validation loss
2023-04-13 16:00:05,659 INFO [trainer.py:650] (6/8) Reaches end of dataloader.
2023-04-13 16:00:05,660 INFO [trainer.py:626] (6/8) Computing validation loss

I need to wait until these calculating is finished? or errors?
Please let me know about the solution.
Thanks

**Comment by lifeiteng** [comment]
The dataset is empty?

what's the output of `python3 ./bin/display_manifest_statistics.py`

**Comment by fullstack-dev0727** [comment]
Thanks for your reply.
Please check attached screenshot for above display_manifest_statistics.py result. 
https://drive.google.com/file/d/1UmSlA58Xt0bMpw3NvsswvtT9-8VIliIB/view?usp=share_link

**Comment by lifeiteng** [comment]
@fullstack-dev0727 what's your problem?

**Comment by lifeiteng** [comment]
> Thanks for your reply. Please check attached screenshot for above display_manifest_statistics.py result. https://drive.google.com/file/d/1UmSlA58Xt0bMpw3NvsswvtT9-8VIliIB/view?usp=share_link

READ the loggings blow, go to the lines around `trainer.py:650`, then debug them.
```
2023-04-13 16:00:05,636 INFO [trainer.py:650] (4/8) Reaches end of dataloader.
2023-04-13 16:00:05,637 INFO [trainer.py:626] (4/8) Computing validation loss
2023-04-13 16:00:05,659 INFO [trainer.py:650] (6/8) Reaches end of dataloader.
2023-04-13 16:00:05,660 INFO [trainer.py:626] (6/8) Computing validation loss
```

**Comment by erogol** [comment]
set drop-last true. It should fix the issue.

**Comment by fullstack-dev0727** [comment]
Hi, @erogol 
Thanks for your message.
But where can I find drop-last option?

**Comment by fullstack-dev0727** [comment]
Hi, @lifeiteng 
I used same command line on stage branch.

python3 bin/trainer.py --max-duration 40 --filter-min-duration 0.5 --filter-max-duration 14 --model-name "VALL-E" --norm-first true --add-prenet false --dtype "float32" --decoder-dim 1024 --nhead 16 --num-decoder-layers 12 --prefix-mode 1 --base-lr 0.05 --warmup-steps 200 --average-period 0 --accumulate-grad-steps 1 --num-epochs 201 --start-epoch 1 --start-batch 0 --exp-dir exp/valle

Btw it seems to be working continuously.
Would you like to let me know about the reason?
I want to train with the latest repo.
Thanks

**Comment by fullstack-dev0727** [comment]
Hi, @lifeiteng 
With one GPU it works.
Would you like to let me know how to fix the issues with multi-GPUs?

**Comment by lifeiteng** [comment]
@fullstack-dev0727 try this https://github.com/lifeiteng/vall-e/pull/93

**Comment by fullstack-dev0727** [comment]
Hi, @lifeiteng 
Thanks for your help.
Training with 8 x GPUs is working with your changes.

**Comment by RuntimeRacer** [comment]
@lifeiteng I actually experience the same problem if I try to train with more than 1 GPU. My AI Training Server has 5x RTX 3090; I was able to train fine, model seemed to be converging as it should, but then apparently hung up at 20k steps this morning while leaving my GPUs run at 100% the whole day without doing anything. :,-)

I tried with `world-size 5` and also `world-size 4`, but it would always get stuck forever at `Computing validation loss` (https://github.com/lifeiteng/vall-e/blob/main/valle/bin/trainer.py#LL793C31-L793C56), which takes like 1 minute if I run it with a single GPU.

I was also able to figure out that I can get further if I comment out these lines: https://github.com/lifeiteng/vall-e/blob/main/valle/bin/trainer.py#L563-L564
Then it's able to calculate validation loss per single GPU and also write to a tensorboard event

But still it gets stuck after that with GPUs at 100% but not continuing training. I am using your provided command from the readme.
Now training on a single GPU because of that for now.

Do you have any suggestions?

**Comment by lifeiteng** [comment]
@chenjiasheng  can you make a PR to fix this?
